src/domain/githubWebhook.go

```diff
@@ -176,8 +177,8 @@ func (h *Handler) GithubWebhook(ctx context.Context, req GithubWebhookRequest) (
 			}
 		}
 
-		if repo.Private != req.Repository.Private || repo.DefaultBranch != req.Repository.DefaultBranch {
-			repo, err = h.db.UpdateRepoPrivateFlagAndDefaultBranch(ctx, req.Repository.DefaultBranch, req.Repository.Private, req.Repository.ID)
+		if repo.Private != req.Repository.Private || repo.Branch != req.Repository.Branch {
```

Comment:
we can drop this piece at all. 
the idea that it's a problem to handle updates from github and it's easier to request this data from a user. so we can remove this piece and remove this function from data store